src/classes/tmControllerDelTrialFeatureRequest.cls

```diff
@@ -0,0 +1,51 @@
+public with sharing class tmControllerDelTrialFeatureRequest {
+	public tmTrialFeatureRequest__c tmTFR { get; private set; }
+  
+  	@TestVisible
+  	public tmDomainTrialFeatureRequest tfrDomain { private set;
+        get {
+            if (tfrDomain == null) {
+                tfrDomain = new tmDomainTrialFeatureRequest(tmTFR.Account__c, tmTFR.Id);
+            } 
+            return tfrDomain;
+        }
+    }
+
+  	public tmControllerDelTrialFeatureRequest (ApexPages.StandardController controller) {
+    	if (!Test.isRunningTest()) controller.addFields(new String[]{'Account__c'});
+    	tmTFR = (tmTrialFeatureRequest__c)controller.getRecord();
+    	isError = false;
+  	}
+
+  	public Boolean isError { get; private set; }
+
+  	public PageReference canDelete() {
+  		isError = true;
+    	PageReference returnPageRef;
+    	try {
+      		if(tfrDomain.tfrDecorator.canDelete) {
+        		Id tmTFRId = ApexPages.currentPage().getParameters().get('id');
+        		tmTrialFeatureRequest__c tmTFRObj = new tmTrialFeatureRequest__c(Id = tmTFRId);
```

Comment:
Here is a security violation that should have been noted. This should be using String.escapeSingleQuotes because it's not safe to accept input over the URL even if it's an ID value.